src/DIRAC/Core/Utilities/JDL.py

```diff
@@ -157,3 +163,161 @@ def dumpCFGAsJDL(cfg, level=1, tab="  "):
+
+        if jdl.lookupAttribute("BannedSites"):
+            jobDescription.bannedSites = set(jdl.getListFromExpression("BannedSites"))
+            jdl.deleteAttribute("BannedSites")
+
+        if jdl.lookupAttribute("JobConfigArgs"):
+            jobDescription.jobConfigArgs = str(jdl.getAttributeString("JobConfigArgs"))
+            jdl.deleteAttribute("JobConfigArgs")
+
+        if jdl.lookupAttribute("CPUTime"):
+            jobDescription.cpuTime = int(jdl.getAttributeInt("CPUTime"))
+            jdl.deleteAttribute("CPUTime")
+
+        if jdl.lookupAttribute("Executable"):
+            jobDescription.executable = jdl.getAttributeString("Executable")
+            jdl.deleteAttribute("Executable")
+
+        if jdl.lookupAttribute("ExecutionEnvironment"):
+            executionEnvironment = jdl.getListFromExpression("ExecutionEnvironment")
+            if executionEnvironment:
+                jobDescription.executionEnvironment = {}
+                for element in executionEnvironment:
+                    key, value = element.split("=")
+                    if value.isdigit():
+                        value = int(value)
+                    else:
+                        try:
+                            value = float(value)
+                        except ValueError:
+                            pass
+                    jobDescription.executionEnvironment[key] = value
+            jdl.deleteAttribute("ExecutionEnvironment")
+
+        if jdl.lookupAttribute("GridCE"):
+            jobDescription.gridCE = jdl.getAttributeString("GridCE")
+            jdl.deleteAttribute("GridCE")
+
+        if jdl.lookupAttribute("InputData"):
+            jobDescription.inputData = jdl.getListFromExpression("InputData")
+            jdl.deleteAttribute("InputData")
+
+        if jdl.lookupAttribute("InputDataPolicy"):
+            jobDescription.inputDataPolicy = jdl.getAttributeString("InputDataPolicy")
+            jdl.deleteAttribute("InputDataPolicy")
+
+        if jdl.lookupAttribute("InputSandbox"):
+            jobDescription.inputSandbox = jdl.getListFromExpression("InputSandbox")
+            jdl.deleteAttribute("InputSandbox")
+
+        if jdl.lookupAttribute("JobGroup"):
+            jobDescription.jobGroup = jdl.getAttributeString("JobGroup")
+            jdl.deleteAttribute("JobGroup")
+
+        if jdl.lookupAttribute("JobName"):
+            jobDescription.jobName = jdl.getAttributeString("JobName")
+            jdl.deleteAttribute("JobName")
+
+        if jdl.lookupAttribute("JobType"):
+            jobDescription.jobType = jdl.getAttributeString("JobType")
+            jdl.deleteAttribute("JobType")
+
+        if jdl.lookupAttribute("LogLevel"):
+            jobDescription.logLevel = jdl.getAttributeString("LogLevel")
+            jdl.deleteAttribute("LogLevel")
+
+        if jdl.lookupAttribute("NumberOfProcessors"):
+            jobDescription.maxNumberOfProcessors = jdl.getAttributeInt("NumberOfProcessors")
```

Comment:
Same question as above with the `int` here: should we convert the result of `getAttributeInt()` into an `int`?
It seems like it is already done by `getAttributeInt()`.
Also the result of `getAttributeInt` might be `None`. Have you tried to set `CPUTime` as `"azerty"`, or `None`, for instance? I guess it will return an exception.